thrust/system/cuda/detail/trivial_copy.inl

```diff
@@ -66,6 +66,16 @@ template<typename System1,
   return cudaMemcpyHostToDevice;
 } // end cuda_memcpy_kind()
 
+#if defined(_WIN32) && !defined(_WIN64)
```

Comment:
Hmm. I misunderstood what the `#if` was guarding. I thought it was surrounding the entire function. Now I see that it is surrounding the implementation.

I'm afraid I don't understand the implementation of this overload. If we're copying from CUDA -> CUDA, why wouldn't the result of this function always be `cudaMemcpyDeviceToDevice`?